storage/key_test.go

```diff
@@ -0,0 +1,145 @@
+package storage
+
+import "testing"
+
+type keyTest struct {
+	Name        string
+	Keyable     Keyable
+	KeyerMethod func() string
+	Expected    string
+}
+
+type keyableFixture struct {
+	kind      string
+	group     string
+	version   string
+	namespace string
+	name      string
+}
+
+func (f keyableFixture) GetKind() string {
+	return f.kind
+}
+
+func (f keyableFixture) GetGroup() string {
+	return f.group
+}
+
+func (f keyableFixture) GetVersion() string {
+	return f.version
+}
+
+func (f keyableFixture) GetName() string {
+	return f.name
+}
+
+func (f keyableFixture) GetNamespace() string {
+	return f.namespace
+}
+
+type noNamespaceKeyableFixture struct {
+	keyableFixture
+}
+
+func (noNamespaceKeyableFixture) NoNamespace() {}
+
+func TestRESTKey(t *testing.T) {
+	fixture := keyableFixture{
+		kind:      "frogs",
```

Comment:
While working on the generic API, I went down the rabbit hole and realized we have a problem here. Our resources kind are singular, not plural. It means that the REST/store keys in this current form are `[...]/check/foo` instead of `[...]/checks/foo`.

I tried to figure out how K8s does it and I'm still confused.

They have a [`Add` method](https://github.com/kubernetes/kubernetes/blob/7f23a743e8c23ac6489340bbb34fa6f1d392db9d/staging/src/k8s.io/apimachinery/pkg/api/meta/restmapper.go#L96) to register types, which uses a function said to be [broken and unsafe](https://github.com/kubernetes/kubernetes/blob/7f23a743e8c23ac6489340bbb34fa6f1d392db9d/staging/src/k8s.io/apimachinery/pkg/api/meta/restmapper.go#L123) to determine the plural of a kind. There's also a [ResourceSingularizer](https://github.com/kubernetes/kubernetes/blob/7f23a743e8c23ac6489340bbb34fa6f1d392db9d/staging/src/k8s.io/apimachinery/pkg/api/meta/restmapper.go#L149:29) function that doesn't seem to be used anywhere execepted in a test file...

Halp

Reply:
Woof! I'll resolve this in a follow-up PR.